File: post-tweet.py
import twitter
from local_settings import *


tweet_text = "Flint, Michigan still doesn't have clean water."


def connect():
    api = twitter.Api( consumer_key=MY_CONSUMER_KEY,
                       consumer_secret=MY_CONSUMER_SECRET,
                       access_token_key=MY_ACCESS_TOKEN_KEY,
                       access_token_secret=MY_ACCESS_TOKEN_SECRET )
    return api


if DEBUG == False:
    status = api.PostUpdate(tweet_text)
    print status.text.encode('utf-8')
else
    print tweet_text

File: post-tweet.js
const twitter = require('twitter');
const moment = require('moment');

var client = new twitter({
  consumer_key: process.env.TWITTER_CONSUMER_KEY,
  consumer_secret: process.env.TWITTER_CONSUMER_SECRET,
  access_token_key: process.env.TWITTER_ACCESS_TOKEN_KEY,
  access_token_secret: process.env.TWITTER_ACCESS_TOKEN_SECRET
});

function numberWithCommas(x) {
    return x.toString().replace(/\B(?=(?:\d{3})+(?!\d))/g, ",");
}

// April 25, 2014
var start = moment([2014, 4, 25]);
var today = moment();
var dayOfWeek = moment().format('dddd');

var since = numberWithCommas( today.diff(start, 'days') );

var text = "Flint, Michigan hasn't had clean water for " + since + " days.";

if ( dayOfWeek === 'Tuesday' ){
        
    // client.post('statuses/update', {status: text},  function(error, tweet, response) {
    //   if(error) throw error;
    //   console.log(tweet);
    //   console.log(response);
    // });
    
    console.log(text);
    
} else {
    
    console.log('Today is not Tuesday.');

}
